Testscripts/Windows/VERIFY-XDP-ACTION.ps1

```diff
@@ -0,0 +1,165 @@
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+        $installXDPCommand = @"
+bash ./XDPDumpSetup.sh 2>&1 > ~/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand
+        Copy-RemoteFiles -uploadTo $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -files "$constantsFile,$LogDir\StartXDPSetup.sh" `
+            -username $user -password $password -upload -runAsSudo
+
+        # Start XDP Setup
+        $testJob = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "bash ./StartXDPSetup.sh" `
+            -RunInBackground -runAsSudo
+        # Terminate process if ran more than 5 mins
+        # TODO: Check max installation time for other distros when added
+        $timer = 0
+            while ((Get-Job -Id $testJob).State -eq "Running") {
+            $currentStatus = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+                -username $user -password $password -command "tail -2 ~/xdpConsoleLogs.txt | head -1" -runAsSudo
+            Write-LogInfo "Current Test Status: $currentStatus"
+            Wait-Time -seconds 20
+            $timer += 1
+            if ($timer -gt 15) {
+                Throw "XDPSetup did not stop after 5 mins. Please check xdpConsoleLogs."
+            }
+        }
+
+        $currentState = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "cat state.txt" -runAsSudo
+
+        if ($currentState -imatch "TestCompleted") {
+            # Start XDP Action Validation
+            Write-LogInfo "Starting XDP validation for Action: $($currentTestData.TestParameters.param)"
+            $testJobDrop = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+                -username $user -password $password -command "./XDP-Action.sh > ~/xdpAction.log" `
+                -RunInBackground -runAsSudo
+            $timer = 0
+            while ((Get-Job -Id $testJobDrop).State -eq "Running") {
+                $currentStatus = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+                    -username $user -password $password -command "tail -2 ~/xdpAction.log | head -1" -runAsSudo
+                Write-LogInfo "Current Test Status: $currentStatus"
+                Wait-Time -seconds 20
+                $timer += 1
+                if ($timer -gt 15) {
+                    Throw "XDPSetup did not stop after 5 mins. Please check logs."
+                }
+            }
+            $currentState = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+                -username $user -password $password -command "cat state.txt" -runAsSudo
+        }
+
+        if ($currentState -imatch "TestCompleted") {

```

Comment:
Dont need this big condition. Use helped function
Get-FinalResultHeader -resultarr $currentState

Reply:
I checked this function and there is no condition for TestCompleted. It is a nice suggestion though I can follow-up with another PR as changes would be taken in other scripts too.